app/lib/katello/repo_discovery.rb

```diff
@@ -1,206 +1,33 @@
 require 'uri'
+require 'spidr'
 
 module Katello
   class RepoDiscovery
     include Katello::Util::HttpProxy
 
-    attr_reader :found, :crawled, :to_follow
+    def self.create_for(content_type = 'yum')
+      class_for(content_type)
+    end
 
-    # rubocop:disable Metrics/ParameterLists
-    def initialize(url, content_type = 'yum', upstream_username = nil,
-      upstream_password = nil, search = '*', crawled = [],
-      found = [], to_follow = [])
-      @uri = uri(url)
-      @content_type = content_type
-      @upstream_username = upstream_username.empty? ? nil : upstream_username
-      @upstream_password = upstream_password.empty? ? nil : upstream_password
-      @search = search
-      @found = found
-      @crawled = crawled
-      @to_follow = to_follow
+    def self.class_for(content_type)
+      case content_type
+      when 'file'
```

Comment:
Can we leverage this => https://github.com/Katello/katello/tree/master/lib/katello/repository_types 
Let the content type tell you the discovery class